save-agent/src/commonMain/kotlin/com/saveourtool/save/agent/utils/FileUtils.kt

```diff
@@ -0,0 +1,74 @@
+/**
+ * Utility methods to work with files using Okio
+ */
+
+package com.saveourtool.save.agent.utils
+
+import com.saveourtool.save.core.files.findAllFilesMatching
+import okio.FileSystem
+import okio.Path
+
+internal expect val fs: FileSystem
+
+/**
+ * Extract path as ZIP archive to provided directory
+ *
+ * @param targetPath
+ */
+internal expect fun Path.extractZipTo(targetPath: Path)
+
+/**
+ * Write content of [this] into a file [file]
+ *
+ * @receiver [ByteArray] to be written into a file
+ * @param file target [Path]
+ * @param mustCreate will be passed to Okio's [FileSystem.write]
+ */
+internal expect fun ByteArray.writeToFile(file: Path, mustCreate: Boolean = true)
+
+/**
+ * Mark [this] file as executable. Sets permissions to rwxr--r--
+ */
+internal expect fun Path.markAsExecutable()
+
+/**
+ * Read file as a list of strings
+ *
+ * @param filePath a file to read
+ * @return list of string from file
+ */
+internal expect fun readFile(filePath: String): List<String>
```

Comment:
why we need it different for linux and java -- maybe user okio.Path

Reply:
Yeah, thanks